Bug 146043589 reindex solr

jsimpson-cc opened the pull request:
[#146043589]

Reindex sunspot when solr server starts.

jsimpson-cc committed:
Call API to run sunspot reindex when solr service starts.

[#146043589]

jsimpson-cc committed:
Call portal API to re-index sunspot after docker container starts and solr server comes up..

[#146043589]

jsimpson-cc committed:
Only attempt re-index if the service API env variable is configured.

[#146043589]

scytacki commented:
Thanks for working on this. I was just about to look at it myself, because I now think we need to fix it before releasing to production.

One change to this is to use an existing api key system in the portal, so that we don't need to add a new one.  I'll comment in the portal side of things: https://github.com/concord-consortium/rigse/pull/338

scytacki committed:
changed this to be notifying the portal

Much of the code from sunspot-reindex.sh was kept.
I tested to make sure the curl commands would fail as expected
when the portal or solr was not up het.

[#146043589]

scytacki commented:
@jsimpson-cc I've updated this to work with the new portal endpoint. And I also tested it to make sure it would keep retrying until my local proxy server found the portal app server. I'm hopefully the AWS load balancer will work the same way.  

I just realized that during an upgrade, the old portal instance might still be around when the solr container comes up.  However I think that will actually work since the first time we apply this, the old portal instance will return an error because it doesn't support the endpoint. So then the script will keep re-trying until the new portal instance is available.

jsimpson-cc reviewed (commented):
Looks good.
